Sources/Xcodeproj/pbxproj().swift

```diff
@@ -94,26 +94,78 @@ public func pbxproj(srcroot: AbsolutePath, projectRoot: AbsolutePath, xcodeprojP
         // Make an array of all the source file reference ids to add to the main group.
         let sourceRefIds = sourceFileRefs.map{ $0.refId }
 
-        // the “Project Navigator” group for this module
-        print("        \(module.groupReference) = {")
-        print("            isa = PBXGroup;")
-        print("            name = '\(module.name)';")
-        print("            path = '\(moduleRoot.relative(to: projectRoot).asString)';")
-        print("            sourceTree = '<group>';")
-        print("            children = (" + sourceRefIds.joined(separator: ", ") + ");")
-        print("        };")
+
+        var groupsChildren = [AbsolutePath: Set<String>]()
+        var rootGroupRefs  = Set<String>()
+        var rootSourcesRefs = [String]()
 
         // the contents of the “Project Navigator” group for this module
         for fileRef in sourceFileRefs {
             let path = fileRef.path.relative(to: moduleRoot)
             print("        \(fileRef.refId) = {")
+            print("            name = \(fileRef.path.basename);")
             print("            isa = PBXFileReference;")
             print("            lastKnownFileType = \(module.fileType(forSource: path));")
-            print("            path = '\(fileRef.path.relative(to: moduleRoot).asString)';")
+            print("            path = '\(fileRef.path.relative(to: srcroot).asString)';")
+            print("            sourceTree = '<group>';")
+            print("        };")
+
+            // If this source file is directly under the module directory
+            if path.dirname == "." {
+                rootSourcesRefs.append(fileRef.refId)
+                continue
+            }
+
+
+            // Generate paths as follows:
+            // Example: 
+            //    input: "MyModule/Foo/foo.swift"
+            //   output: ["MyModule", 
+            //            "MyModule/Foo"]
+            //
+            var paths = [AbsolutePath]()
```

Comment:
I think something like:
```
let paths = [AbsolutePath](sequence(first: path.parentDirectory, next: { path in 
   return path == AbsolutePath.root ? nil : path.parentDirectory
}))
```
will accomplish the same thing, but maybe more readably. In fact, that particularl sequence is useful enough that it maybe should be part of `AbsolutePath`.